r/aws_launch_template: Crash in `TestAccEC2LaunchTemplate_licenseSpecification`

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

In nightly CI:

```
=== RUN   TestAccEC2LaunchTemplate_licenseSpecification
=== PAUSE TestAccEC2LaunchTemplate_licenseSpecification
=== CONT  TestAccEC2LaunchTemplate_licenseSpecification
------- Stderr: -------
panic: interface conversion: interface {} is *schema.Set, not []interface {}
goroutine 397 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.expandRequestLaunchTemplateData(0x96cfcde)
  /opt/teamcity-agent/work/87cd0779df5e20c9/internal/service/ec2/ec2_launch_template.go:983 +0x1c54
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.resourceLaunchTemplateCreate(0x0, {0x82327a0, 0xc0005fca80})
  /opt/teamcity-agent/work/87cd0779df5e20c9/internal/service/ec2/ec2_launch_template.go:714 +0x1e7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xa88abe0, {0xa88abe0, 0xc002154180}, 0xd, {0x82327a0, 0xc0005fca80})
  /var/lib/teamcity/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.14.0/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000b9ec40, {0xa88abe0, 0xc002154180}, 0xc001bbaa90, 0xc002170180, {0x82327a0, 0xc0005fca80})
  /var/lib/teamcity/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.14.0/helper/schema/resource.go:837 +0xc29
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00697c558, {0xa88ab38, 0xc0069e2980}, 0xc00213c5a0)
  /var/lib/teamcity/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.14.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002e3e1e0, {0xa88abe0, 0xc00213fc50}, 0xc00208bea0)
  /var/lib/teamcity/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:812 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x9443e60, 0xc002e3e1e0}, {0xa88abe0, 0xc00213fc50}, 0xc002144300, 0x0)
  /var/lib/teamcity/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00195a700, {0xa99ae78, 0xc000eca1a0}, 0xc002150000, 0xc002b099b0, 0x10726a60, 0x0)
  /var/lib/teamcity/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc00195a700, {0xa99ae78, 0xc000eca1a0}, 0xc002150000, 0x0)
  /var/lib/teamcity/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
  /var/lib/teamcity/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
  /var/lib/teamcity/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:919 +0x294
```

https://github.com/hashicorp/terraform-provider-aws/blob/a0d6930dbd9a03d0019f746e550340c885824e21/internal/service/ec2/ec2_launch_template.go#L983-L985

--- a/.changelog/24579.txt
+++ b/.changelog/24579.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_luanch_template: Fix crash when reading `license_specification`
+```

--- a/.changelog/24579.txt
+++ b/.changelog/24579.txt
@@ -1,3 +1,3 @@
 ```release-note:bug
-resource/aws_luanch_template: Fix crash when reading `license_specification`
+resource/aws_launch_template: Fix crash when reading `license_specification`
 ```

--- a/internal/service/ec2/ec2_launch_template.go
+++ b/internal/service/ec2/ec2_launch_template.go
@@ -980,8 +980,8 @@ func expandRequestLaunchTemplateData(d *schema.ResourceData) *ec2.RequestLaunchT
 		apiObject.KeyName = aws.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("license_specification"); ok && len(v.([]interface{})) > 0 {
-		apiObject.LicenseSpecifications = expandLaunchTemplateLicenseConfigurationRequests(v.([]interface{}))
+	if v, ok := d.GetOk("license_specification"); ok && v.(*schema.Set).Len() > 0 {
+		apiObject.LicenseSpecifications = expandLaunchTemplateLicenseConfigurationRequests(v.(*schema.Set).List())
 	}
 
 	if v, ok := d.GetOk("maintenance_options"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {

--- a/internal/service/ec2/ec2_launch_template_test.go
+++ b/internal/service/ec2/ec2_launch_template_test.go
@@ -1300,7 +1300,10 @@ func TestAccEC2LaunchTemplate_licenseSpecification(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { acctest.PreCheck(t) },
+		PreCheck: func() {
+			acctest.PreCheck(t)
+			acctest.PreCheckIAMServiceLinkedRole(t, "/aws-service-role/license-manager.amazonaws.com")
+		},
 		ErrorCheck:   acctest.ErrorCheck(t, ec2.EndpointsID),
 		Providers:    acctest.Providers,
 		CheckDestroy: testAccCheckLaunchTemplateDestroy,

